### PR TITLE
perf: avoid intermediate calls for Colors.FromInteger

### PR DESCRIPTION
**Pull request by spouliot**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

- Refactoring (no functional changes, no api changes)

## What is the current behavior?

```
Colors.FromInteger(int) ->
	Colors.FromARGB(byte, byte, byte, byte) ->
		Color.FromArgb(byte, byte, byte, byte) ->
			new Color(uint)
```

```
BenchmarkDotNet=v0.13.4, OS=macOS 13.0.1 (22A400) [Darwin 22.1.0]
Apple M1, 1 CPU, 8 logical and 8 physical cores
.NET SDK=7.0.100
  [Host]     : .NET 7.0.0 (7.0.22.51805), Arm64 RyuJIT AdvSIMD
  DefaultJob : .NET 7.0.0 (7.0.22.51805), Arm64 RyuJIT AdvSIMD

|      Method |     Mean |   Error |  StdDev |
|------------ |---------:|--------:|--------:|
| FromInteger | 327.9 us | 0.80 us | 0.71 us |
```

## What is the new behavior?

```
FromInteger(int) -> new Color(uint)
```

```
BenchmarkDotNet=v0.13.4, OS=macOS 13.0.1 (22A400) [Darwin 22.1.0]
Apple M1, 1 CPU, 8 logical and 8 physical cores
.NET SDK=7.0.100
  [Host]     : .NET 7.0.0 (7.0.22.51805), Arm64 RyuJIT AdvSIMD
  DefaultJob : .NET 7.0.0 (7.0.22.51805), Arm64 RyuJIT AdvSIMD

|      Method |     Mean |   Error |  StdDev |
|------------ |---------:|--------:|--------:|
| FromInteger | 327.6 us | 0.21 us | 0.17 us |
```

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

The net7 JIT compiler will inline them… but not the interpreter [1].

Still at least the metadata will be present as the linker won’t be able to remove the extra methods. It's also easier to step in while debugging the code.

[1] https://github.com/spouliot/benchmarks/tree/main/Colors_FromInteger

Internal Issue (If applicable):

### CODE REVIEW

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UWP/UI/Color.cs [hunk 1]
```diff
@@ -46,6 +46,16 @@ private Color(byte a, byte r, byte g, byte b)
...
+			_b = 0;
+			_g = 0;
+			_r = 0;
+			_a = 0;
+			_color = color;
+		}
+
 		public override bool Equals(object o) => o is Color color && Equals(color);
 
 		public bool Equals(Color color) =>
```
[the remark this answers is not in the document]
Comment (spouliot) [reply src/Uno.UWP/UI/Color.cs:61]: It's the `[FieldOffset]` attributes. Both the `color` (4 bytes) and the 4 bytes `a`, `r`, `g` and `b` share the same memory space. 

IOW updating `color` updates `a`, `r`, `g` and `b` - just like updating any of `a`, `r`, `g` and `b` updates `color`

File: src/Uno.UWP/UI/Colors.cs [hunk 2]
```diff
@@ -16,20 +16,9 @@ static partial class Colors
...
-		public static Color FromInteger(int color)
-		{
-			return FromARGB(
-				(byte)((color & 0xFF000000) >> 24),
-				(byte)((color & 0x00FF0000) >> 16),
-				(byte)((color & 0x0000FF00) >> 8),
-				(byte)((color & 0x000000FF))
-			);
-		}
+		public static Color FromInteger(int color) => new((uint)color);
```
Comment (Youssef1313) [inline src/Uno.UWP/UI/Colors.cs:21]: This change can cause regressions. Previously the produced `Color` would have correct values for `A`, `R`, `G`, and `B` properties. Now they will be zero.

Comment (Youssef1313) [reply src/Uno.UWP/UI/Colors.cs:21]: Ignore this comment. I just saw your comment above.

File: src/Uno.UWP/UI/Color.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/Uno.UWP/UI/Color.cs:61]: Oh nice!! That is very smart!

**Review by Youssef1313 — APPROVED** [review]

**Pull request merged**